enhancements/machine-api/control-plane.md

```diff
@@ -0,0 +1,281 @@
+- To managed individual Control Plane components. Self hosted Control Plane components that are self managed by their operators is an assumed invariant.
+- To integrate with any Control Plane components topology e.g Remote hosted pod based Control Plane.
+- Automated disaster recovery of a cluster that has lost quorum.
+- To manage OS upgrades. This is managed by the Machine Config Operator.
+- To manage configuration changes at the software layer. This is managed by the Machine Config Operator.
+- To automate the provisioning and decommission of the bootstrapping instance managed by the installer.
+
+## Proposal
+
+This proposes an ad-hoc CRD and controller for declaratively managing as a single entity the compute resources that host the OCP Control Plane components.
+
+This entity is decoupled and orthogonal to the lifecycle and management of the Control Plane components that it hosts:
+- Rolling OS upgrades and config changes at the software layer are managed by the Machine Config Operator.
+- Etcd Guard that ensures a PDB to honour quorum is managed by the Machine Config Operator.
+- Each individual Control Plane component i.e Kube API Server, Scheduler and controller manager are self hosted and managed by their operators.
+- The Cluster Etcd Operator manages certificates, report healthiness and add Etcd members as "Master" nodes join the cluster.
+- The Kubelet is tied to the OS and therefore is managed by the Machine Config Operator.
+
+All of these components are expected to keep self managing themselves as the cluster shrink and expand the Control Plane compute resources.
+
+### User Stories [optional]
+
+#### Story 1
+- As an operator running Installer Provider Infrastructure (IPI), I want flexibility to run [large or small clusters](https://kubernetes.io/docs/setup/best-practices/cluster-large/#size-of-master-and-master-components) so I need to have the ability to resize the control plane in a declarative, automated and seamless manner.
+
+#### Story 2
+- As an operator running User Provider Infrastructure (UPI), I want to expose my non-machine API machines and offer them to the Control Plane controller so I can have the ability to resize the control plane in a declarative, automated and seamless manner.
+
+#### Story 3
+- As an operator of an OCP Dedicated Managed Platform, I want to give users flexibility to add as many workers nodes as they want or to enable autoscaling on woker nodes so I need to have ability to resize the control plane instances in a declarative, automated and seamless manner to react quickly to cluster growth.
+
+#### Story 4
+- As a SRE, I want to have consumable API primitives in place to resize Control Plane compute resources so I can develop upper level automation tooling atop. E.g Control Plane autoresizing to support a severe growth peak of the number of worker nodes.
```

Comment:
Yes, I'd like to focus this proposal on settling the right primitives to enable that and leave the discussion around when/what should autoscale the replica field for a follow up RFE that builds on top of this.

Reply:
I agree that this proposal is to create right primitives, but i couldn't find an mention of how these primitives will enable the https://github.com/openshift/enhancements/pull/278#discussion_r404368590

So if you  can include a new lines on how you see the current proposal will full that, it would be very helpful

Reply:
Covered here https://github.com/openshift/enhancements/pull/278/commits/3b388438df054376d7d7ddd6c3ea2846d5501208